src/compiler/checker.ts

```diff
@@ -4379,6 +4379,8 @@ module ts {
                     }
                 }
 
+                Debug.assert(!!downfallType, "If there is no common supertype, each type should have a downfallType");

```

Comment:
downfallType? Is this supposed to be like a fallbackType? Rename it so. downfallType sounds like it encompasses revolutions, lost wars, etc.